sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategy.scala

```diff
@@ -130,7 +130,9 @@ private[sql] object FileSourceStrategy extends Strategy with Logging {
           createNonBucketedReadRDD(readFile, selectedPartitions, fsRelation)
       }
 
+      // These metadata values make scan plans uniquely identifiable for equality checking.
       val meta = Map(
```

Comment:
not related to this pr - i think we should remove the concept of metadata entirely from physical plans. It is something added as a hack to propagate the following information, which really should just be a named field in those case classes.